src/vehicle_gui.cpp

```diff
@@ -1395,7 +1395,36 @@ void BaseVehicleListWindow::DrawVehicleListItems(VehicleID selected_vehicle, int
 		DrawVehicleImage(v, image_left, image_right, y + FONT_HEIGHT_SMALL - 1, selected_vehicle, EIT_IN_LIST, 0);
 		DrawString(text_left, text_right, y + line_height - FONT_HEIGHT_SMALL - WD_FRAMERECT_BOTTOM - 1, STR_VEHICLE_LIST_PROFIT_THIS_YEAR_LAST_YEAR);
 
-		if (!v->name.empty()) {
+		/* Get the cargoes the vehicle can carry */
+		CargoTypes vehicle_cargoes = 0;
```

Comment:
variables can go inside the if-scope now I guess :)

Reply:
Indeed. I tried whether I can put the string concatenation part into the inner branches, but it was starting to be messy and always crashed the game, so I gave up and stayed with this solution. :)